### ISSUE TITLE
.dependencies in lisp/Makefile

### ISSUE THREAD

**Issue by jabranham** [emacs-ess/ESS#768]
@vspinu the new .dependencies target in lisp/Makefile doesn't account for if the .el files are in subfolders like obsolete/. How do you want to tackle this?

So for example if you check out #732, `.dependencies` has this line: `obsolete/ess-noweb.elc: ess-noweb-mode.elc` which won't work.

**Comment by vspinu** [comment]
Right. But do we want to to bother with those? The obsoleted files will be rarely modified if at all, so extra complexity in dependency generation doesn't seem warranted.

**Comment by jabranham** [comment]
`make all` seems to fail, see the travis error in #732. We could modify .dependencies so it doesn't recurse into obsolete (and manually include the files in obsolete/ in the makefile, I guess?)

**Comment by vspinu** [comment]
Haven't tested on your branch but should be fine now.

**Comment by mmaechler** [comment]
There's still (or again?) a problem - almost surely with `.dependencies`:  
Currently,  `make lisp` always re-bytecompiles 3 out of the 7  `obsolete/*.el` files

If I look at the `.dependencies` file, I do see too many lines, notably things such as

<foo>.elc : obsolete/<foo>.elc

which "must be wrong"
